src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Debug/LaunchSettingsProvider.cs

```diff
@@ -164,17 +168,38 @@ protected set
             }
         }
 
+        public override NamedIdentity DataSourceKey { get; } = new NamedIdentity(nameof(LaunchSettingsProvider));
+
+        // _nextVersion represents the version we will use in the future, so we need to
+        // subtract 1 to get the current version.
+        public override IComparable DataSourceVersion => _nextVersion - 1;
```

Comment:
Is it guaranteed that `_nextVersion` is always `currentVersion + 1`?

Reply:
Yes.

Note that I would like to fix this up so that we're no longer embedding the version in the `ILaunchSettings` implementation, but getting it from the `IProjectVersionedValue` everywhere we need it. That is also beyond the scope of this change.